app/src/main/java/de/xikolo/download/filedownload/FileDownloadHandler.kt

```diff
@@ -0,0 +1,240 @@
+import com.tonyodev.fetch2core.Func
+import com.tonyodev.fetch2core.Func2
+import de.xikolo.App
+import de.xikolo.download.DownloadHandler
+import de.xikolo.download.DownloadStatus
+import de.xikolo.download.filedownload.FileDownloadRequest.Companion.REQUEST_EXTRA_SHOW_NOTIFICATION
+import de.xikolo.download.filedownload.FileDownloadRequest.Companion.REQUEST_EXTRA_TITLE
+import de.xikolo.utils.NotificationUtil
+
+object FileDownloadHandler : DownloadHandler<FileDownloadIdentifier, FileDownloadRequest> {
+
+    private val disabledNotificationsConfiguration =
+        FetchConfiguration.Builder(App.instance)
+            .setAutoRetryMaxAttempts(1)
+            .setDownloadConcurrentLimit(5)
+            .enableFileExistChecks(false)
+            .setNamespace("XIKOLO_FETCH_NO_NOTIFICATIONS")
+            .build()
+
+    private val enabledNotificationsConfiguration =
+        FetchConfiguration.Builder(App.instance)
+            .setAutoRetryMaxAttempts(1)
+            .setDownloadConcurrentLimit(5)
+            .enableFileExistChecks(false)
+            .setNamespace("XIKOLO_FETCH")
+            .setNotificationManager(object : DefaultFetchNotificationManager(App.instance) {
+                override fun getFetchInstanceForNamespace(namespace: String): Fetch {
+                    return enabledNotificationsManager
+                }
+
+                override fun getDownloadNotificationTitle(download: Download): String {
+                    return download.request.extras.getString(
+                        REQUEST_EXTRA_TITLE,
+                        download.request.url
+                    )
+                }
+
+                override fun shouldCancelNotification(
+                    downloadNotification: DownloadNotification
+                ): Boolean {
+                    return downloadNotification.isCompleted
+                }
+
+                override fun getChannelId(notificationId: Int, context: Context): String {
+                    NotificationUtil(context)
+                    return NotificationUtil.DOWNLOADS_CHANNEL_ID
+                }
+            })
+            .build()
+
+    private val enabledNotificationsManager: Fetch = Fetch.getInstance(
+        enabledNotificationsConfiguration
+    )
+
+    private val disabledNotificationsManager: Fetch = Fetch.getInstance(
+        disabledNotificationsConfiguration
+    )
+
+    init {
+        val listener = object : FetchListener {
+            override fun onAdded(download: Download) {
+                notifyStatus(download)
+            }
+
+            override fun onCancelled(download: Download) {
+                cancel(FileDownloadIdentifier(download.id))
+                notifyStatus(download)
+            }
+
+            override fun onCompleted(download: Download) {
+                if (download.extras.getString(
+                        REQUEST_EXTRA_SHOW_NOTIFICATION,
+                        true.toString()
+                    ) == true.toString()
```

Comment:
That's somewhat arkward, because the `Extras` constructor accepts a `Map<String, String>` as arguments and no `putBoolean` method. But it has the method `getBoolean` whicht tries to convert the String value to a Boolean. I now use the `getBoolean` method but a library change then might break something.

Reply:
Not a blocker, take the most stable variant.